bulk/job.go

```diff
@@ -131,10 +149,11 @@ type FailedRecord struct {
 type Options struct {
 	ColumnDelimiter     ColumnDelimiter `json:"columnDelimiter"`
 	ContentType         ContentType     `json:"contentType"`
-	ExternalIDFieldName string          `json:"externalIdFieldName"`
+	ExternalIDFieldName string          `json:"externalIdFieldName,omitempty"`
 	LineEnding          LineEnding      `json:"lineEnding"`
-	Object              string          `json:"object"`
+	Object              string          `json:"object,omitempty"`
 	Operation           Operation       `json:"operation"`
+	Query               string          `json:"query,omitempty"`
 }
```

Comment:
This structure is not really following SRP.

I would recommended adding a function that takes in the options and then the query specific parameters.

Also you can use `soql QueryFormatter` interface for the query.  This would allow them to use the same query formatter for the SOQL API and the query 2.0 api

Reply:
Any problem to keep that way?
Base on https://developer.salesforce.com/docs/atlas.en-us.api_bulk_v2.meta/api_bulk_v2/query_create_job.htm, **query** is a Request parameters that could be optional. That way is possible to keep the sam eBulk 2.0 struct and just change the *operation* and *parameters*